bson/doc.go

```diff
@@ -19,24 +20,96 @@
 // 		i32, ok := val.Int32OK()
 // 		// do something with i32...
 //
-// The D family of types is used to build concise representations of BSON using native Go types.
-// These types do not support automatic lookup.
+// Native Go Types
+//
+// This D and M types defined in this package can be used to build concise representations of BSON using native Go
```

Comment:
Consider a sentence describing when you definitely need to use D (i.e. for commands)

Reply:
This is described where the types are defined and I'd prefer not to duplicate when possible. Added a note to say users should refer to the type definitions for use cases and more info.